nixos/tests/keymap.nix

```diff
@@ -29,54 +29,72 @@ let
 
   mkKeyboardTest = layout: { extraConfig ? {}, tests }: with pkgs.lib; let
     combinedTests = foldAttrs (acc: val: acc ++ val) [] (builtins.attrValues tests);
-    perlStr = val: "'${escape ["'" "\\"] val}'";
+    perlStr = val: "\n            \"${val}\",";
     lq = length combinedTests.qwerty;
     le = length combinedTests.expect;
     msg = "length mismatch between qwerty (${toString lq}) and expect (${toString le}) lists!";
-    send   = concatMapStringsSep ", " perlStr combinedTests.qwerty;
-    expect = if (lq == le) then concatStrings combinedTests.expect else throw msg;
+    send   = "(" + concatMapStrings perlStr combinedTests.qwerty + "\n        )";
```

Comment:
This is used down in a loop in python. Can't we just do more of that in python, instead of doing it in nix and then interpolating it into python?